main/SCsub

```diff
@@ -7,8 +7,10 @@ import main_builders
 env.main_sources = []
 
 env_main = env.Clone()
-
-env_main.add_source_files(env.main_sources, "*.cpp")
+if env["shared"]:
```

Comment:
main had compilation issues when converting it to shared build, so instead of spending too much time investigating it remains static in the shared build so gains from shared build can begin to be reaped. I will investigate and fix in another PR.

Reply:
now that the modules are all shared this can also be built shared